src/engine/Engine.hpp

```diff
@@ -56,25 +49,16 @@ class Engine {
         return m_activeLayers.find(idx) != m_activeLayers.end();
     }
 
-    inline bool ActivateLayer(size_t idx) {
+    inline void ActivateLayer(size_t idx) {
         if (m_layers.find(idx) != m_layers.end() &&
             !(m_activeLayers.find(idx) != m_activeLayers.end())) {
-            const bool wasInserted = m_activeLayers.insert(idx).second;
-
-            if (wasInserted) {
-                m_layers[idx]->OnShown();
-            }
```

Comment:
Many of these checks if the layer has been added are removed. These checks are still needed

Reply:
Yes, I'll fix this.